From: Parav Pandit <parav@mellanox.com>
Date: Tue, 21 Jul 2020 19:53:52 +0300
Subject: devlink: Avoid duplicate check for reload enabled flag
Patch-mainline: v5.9-rc1
Git-commit: 9232a3e67b212f9ef924786f8dde23080acd321a
References: bsc#1176447

Reload operation is enabled or not is already checked by
devlink_reload(). Hence, remove the duplicate check from
devlink_nl_cmd_reload().

Signed-off-by: Parav Pandit <parav@mellanox.com>
Reviewed-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 net/core/devlink.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/core/devlink.c
+++ b/net/core/devlink.c
@@ -2973,7 +2973,7 @@ static int devlink_nl_cmd_reload(struct
 	struct net *dest_net = NULL;
 	int err;
 
-	if (!devlink_reload_supported(devlink) || !devlink->reload_enabled)
+	if (!devlink_reload_supported(devlink))
 		return -EOPNOTSUPP;
 
 	err = devlink_resources_validate(devlink, NULL, info);
